conbench/entities/commit.py

```diff
@@ -36,6 +42,69 @@ class Commit(Base, EntityMixin):
     def get_parent_commit(self):
         return Commit.first(sha=self.parent, repository=self.repository)
 
+    def get_fork_point_commit(self) -> Optional["Commit"]:
+        if self.sha == self.fork_point_sha:
+            return self
+        else:
+            return Commit.first(sha=self.fork_point_sha, repository=self.repository)
+
+    @property
+    def ancestor_commit_query(self) -> Query:
+        """Return a query that returns the IDs and timestamps of all Commits in the
+        direct ancestry of this commit, ordered starting with this commit, backwards in
+        lineage.
```

Comment:
Good ideas; I just added a comparison to `git log` and an ASCII graph to the docstring.

>do we (intend to) store the (partial) history of a git repo in our database, i.e. the relationship between commits?

Yes! This was the solution we chose to solve https://github.com/conbench/conbench/milestone/1 . (Click "show more" in the description of that milestone to see more context.)

>doing local `git ...` operations (managed by Conbench)

This is a good idea, but it can't deal with git history rewrites well, which happen often in the repos we track. These queries are complex but are able to fully track the temporal relationships between different BenchmarkResults that Conbench stores, even after history rewrites.

Reply:
For Ed's comments:

>maybe commit_ancestry_query or commit_history_query would be more obvious?

Yeah! Since `history` is used in other contexts elsewhere, maybe I'll change to `commit_ancestry_query`

>But maybe a comment or note in the docstring of warning?

Will do. This was intentional because downstream queries that use this as a subquery can choose whether to impose their own limit or not.

>Is there value to adding an index column so order is easily maintained downstream?

Ah, this is a much better idea than copying/pasting the rationale everywhere. 🙂

Reply:
Okay, I reworked `commit_ancestry_query` to return a `commit_order` column that downstream queries may take advantage of.